packages/daxus-next/src/StreamHydration.tsx

```diff
@@ -0,0 +1,97 @@
+'use client';
+
+import { useDatabaseContext } from 'daxus';
+import type { NotifyDatabaseContext } from 'daxus/dist/model/types.js';
+import { InfiniteAccessor } from 'daxus/dist/model/index.js';
+import { useState, type ReactNode, useEffect, useRef } from 'react';
+import { useServerInsertedHTML } from 'next/navigation.js';
+
+interface Props {
+  children: ReactNode;
+}
+
+function isServer() {
+  return typeof window === 'undefined';
+}
+
+function isUndefined(value: unknown): value is undefined {
+  return typeof value === 'undefined';
+}
+
+export function StreamHydration({ children }: Props) {
+  const database = useDatabaseContext();
+  const [trackedCtx] = useState(() => new Set<NotifyDatabaseContext>());
+  if (!database) throw new Error('Should be wrapped with a database provider!');
+
+  // server stuff
+  if (isServer()) {
+    database.subscribe(ctx => {
+      trackedCtx.add(ctx);
+    });
+  }
+
+  const countRef = useRef(0);
+  useServerInsertedHTML(() => {
+    const ctxs: NotifyDatabaseContext[] = [];
+    trackedCtx.forEach(ctx => {
+      ctxs.push(ctx);
+    });
+    trackedCtx.clear();
+
+    if (!ctxs.length) return null;
+
+    const serializedCtxs = ctxs.map(ctx => JSON.stringify(ctx)).join(',');
+
+    countRef.current++;
+    return (
+      <script
+        key={countRef.current}
+        dangerouslySetInnerHTML={{
+          __html: `
+        window['__daxus'] = window['__daxus'] ?? [];
+        window['__daxus'].push(${serializedCtxs})
```

Comment:
```suggestion
        window['__daxus'].push(${htmlEscapeJsonString(serializedCtxs)})
```

You should escape the content otherwise if there is `</script>` in the serializedCtx this may explode the browser.

You can copy the function from next.js:

https://github.com/vercel/next.js/blob/86160a5190c50ea315c7ba91d77dfb51c42bc65f/packages/next/server/htmlescape.ts

Reply:
fixed it in [b967312](https://github.com/jason89521/daxus/pull/191/commits/b967312e3cfc2b2bc7931129bd90f778c8461f1e)